src/main/java/uk/nhs/digital/nhsconnect/nhais/repository/InboundStateRepositoryExtensionsImpl.java

```diff
@@ -0,0 +1,37 @@
+package uk.nhs.digital.nhsconnect.nhais.repository;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.stereotype.Repository;
+import uk.nhs.digital.nhsconnect.nhais.model.mesh.WorkflowId;
+
+import java.util.Optional;
+
+import static org.springframework.data.mongodb.core.query.Criteria.where;
+import static org.springframework.data.mongodb.core.query.Query.query;
+
+@Repository
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class InboundStateRepositoryExtensionsImpl implements InboundStateRepositoryExtensions {
+
+    private final MongoOperations mongoOperations;
+
+    @Override
+    public Optional<InboundState> findBy(
```

Comment:
Crud repo could do that but method name would be really long

Reply:
You could:
````java
@Repository
public interface InboundStateRepository extends CrudRepository<InboundState, String> {

    @Query("{ 'workflowId' : ?0, 'sender' : ?1, 'recipient' : ?2, 'interchangeSequence' : ?3, 'messageSequence' : ?4, 'transactionNumber' : ?5}")
    Optional<InboundState> findBy(
            WorkflowId workflowId, String sender, String recipient, Long interchangeSequence, Long messageSequence, Long transactionNumber);
}
````